src/data/nat/factorization/basic.lean

```diff
@@ -249,33 +249,83 @@ by { cases n, refl }
 lemma factorization_equiv_inv_apply {f : ℕ →₀ ℕ} (hf : ∀ p ∈ f.support, prime p) :
   (factorization_equiv.symm ⟨f, hf⟩).1 = f.prod pow := rfl
 
-/-! ### Factorization and divisibility -/
+/-! ### Generalisation of the "even part" and "odd part" of a natural number -/
 
-lemma dvd_of_mem_factorization {n p : ℕ} (h : p ∈ n.factorization.support) : p ∣ n :=
-begin
-  rcases eq_or_ne n 0 with rfl | hn, { simp },
-  simp [←mem_factors_iff_dvd hn (prime_of_mem_factorization h), factor_iff_mem_factorization.mp h],
-end
+-- The largest power of `p` that divides into `n`.  For `p = 2` this is the even part of `n`.
```

Comment:
Let's make this a section docstring so that it shows up in the documentation. This should be merged into the `/-! ... -/` above.
```suggestion
/-! We introduce the notations `ord_proj[p] n` for the largest power of the prime `p` that
divides `n` and `ord_compl[p] n` for the complementary part. The `ord` naming comes from
the $p$-adic order/valuation of a number, and `proj` and `compl` are for the projection and
complementary projection. The term `n.factorization p` is the $p$-adic order itself.

For example, `ord_proj[2] n` is the even part of `n` and `ord_compl[2] n` is the odd part. -/
```